### PR TITLE
Fix : Secondary Button Transition

### PR DESCRIPTION
**Pull request by Vrishabhsk**
## What? + How?

- Modify `box-shadow` style for `components-button.is-secondary` to allow `transition delay` to be applied
- Reference used to fix it : [`Ref-1`](https://stackoverflow.com/a/25410897) & [`Ref-2`](https://stackoverflow.com/a/22461987)

## Why?

- https://github.com/WordPress/gutenberg/issues/66007
- Transitions don't work between `insets` and `outsets`
> This transition works for the primary/tertiary variations but not the secondary variation.

## Screenshots or screencast 
- `Original`

https://github.com/user-attachments/assets/fda32cc2-7b13-4fa2-82d9-5db55e09403c

- `Fixed`

https://github.com/user-attachments/assets/9003d4ef-9e33-4e0d-bdf5-3042b2b37ea4

### CODE REVIEW

File: packages/components/src/button/style.scss [hunk 1]
```diff
@@ -43,6 +43,10 @@
 		outline: 3px solid transparent;
 	}
 
+	&.is-secondary:focus:not(:disabled) {
+		box-shadow: 0 0 0 $black inset, 0 0 0 var(--wp-admin-border-width-focus) $components-color-accent;
+	}
```
Comment (t-hamano) [inline packages/components/src/button/style.scss:48]: This style would be better off moving it [here](https://github.com/WordPress/gutenberg/pull/66045/files#diff-0688ad494526b3b921c4444dc8bc6f63efc4ad74c7c1123b061490cfc57f293aR154).

Comment (t-hamano) [inline packages/components/src/button/style.scss]: This style would be better off moving it [here](https://github.com/WordPress/gutenberg/pull/66045/files#diff-0688ad494526b3b921c4444dc8bc6f63efc4ad74c7c1123b061490cfc57f293aR154).

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

`$black` looks a bit unnatural to me. Below I temporarily changed the transition-delay from 0.1s to 1s to better demonstrate the focus outline transition:

https://github.com/user-attachments/assets/cfb67617-7b51-48ea-94f7-ec94e3a7e9c1

`currentColor` feels smoother to me:

https://github.com/user-attachments/assets/373fad41-9c3a-4660-a0f7-e90cbd164e00

**Comment by Vrishabhsk** [comment]
Thanks @t-hamano
- `currentColor` does fit in better, and increasing the delay does point out the flaw in using `$black`
- I have modified the PR accordingly : [`0147438`](https://github.com/WordPress/gutenberg/pull/66045/commits/01474380da44593fea02bb86c1fb869b5f57f284)
- Also, do you want me to implement the same for the `feature image button` as mentioned here : [`#65299`](https://github.com/WordPress/gutenberg/issues/65299#issuecomment-2404613573)

Let me know. Thanks

**Review by tyxla — COMMENTED** [review]
Secondary button transition looks good, thanks for fixing it @Vrishabhsk! 🙌 

Before we ship it, could you please add a CHANGELOG entry?

**Comment by tyxla** [comment]
> Also, do you want me to implement the same for the feature image button as mentioned here : #65299

That would be great, thank you! Ideally, that button should just inherit the styles and transitions from the default secondary button. That might mean some style overrides could be cleaned up. 

Feel free to address it in another PR, of course.

**Review by tyxla — APPROVED** [review]
LGTM 👍 ✅ 

Thanks for the fix @Vrishabhsk 🚀

**Pull request merged**